lib/core/note_storage.dart

```diff
@@ -51,6 +52,13 @@ class NoteStorage {
     return catchAll<Note>(() async {
       assert(note.fullFilePath.startsWith(p.separator));
 
+      final directory = dirname(note.fullFilePath);
+      final directoryExists = io.Directory(directory).existsSync();
+      if (!directoryExists) {
```

Comment:
This is necessary to allow new folders to be created for filename templates like this:
```
folder_name/{{date}} {{title}}
```